lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -16,12 +16,13 @@ def run(&block)
           queue = build_queue
           lock = Mutex.new
 
-          @workers = thread_count.times.map do
-            Thread.new do
+          @workers = Array.new(thread_count) do
+            t = Thread.new do
               while (item = next_item(queue, lock))
                 yield item
               end
             end
+            t.abort_on_exception = true
```

Comment:
@ABaldwinHunter RE: @pbrisbin's comment... is it the case that we want different `abort_on_exception` on different threads within this process?